content/news/20.4networking-rework.md

```diff
@@ -0,0 +1,224 @@
+{{< /notice >}}
+
+#### The context
+The context contains information, callbacks and entry points, to access the surrounding network system, the main thread, as well as handling processing other packets, or completing configuration tasks.
+
+##### ReplyHandler
+The reply handler can be used to quickly send a payload back to the sender. It is for example useful to send and answer to a query packet, or simply send an acknowledgment that you received and processed the payload. You still need to register the return payload.
+
+##### PacketHandler
+In case you implement a packet splitting mechanism, whether that splits on full vanilla packets, or custom packet payloads, the `IPacketHandler` interface gives you access to the start of the processing pipeline, allowing you to process other payloads immediately.
+
+{{< notice note >}}
+This does not transmit payloads, it purely allows for the receiving end to process additional packets or payloads constructed in memory during the processing of your payload.
+{{< /notice >}}
+
+The packet handler also gives you access to a `disconnect(Component)` method, allowing you to terminate the connection, and showing the given component as reason to the user.
+
+##### WorkHandler
+The work handler allows you to schedule work on the main thread of the receiving side. This might be the `Minecraft` class instance if the logical receiving side is the client, or the `MinecraftServer` instance if the receiving side is the server.
+
+The system uses `CompleteableFuture` instances, so you can schedule different follow-up tasks, if need be.
+
+##### PacketFlow
+The context will indicate via the packet flow, what the receiving side currently is. If it is a server bound flow, then handler is currently being invoked in the context of the server. Is it a client bound flow, then the handler is currently being invoked in the context of the client.
+
+##### ConnectionProtocol
+The current active connection protocol. Useful if you have raw bytes of a packet wrapped in your payload, allowing your handler to decode the inner packet or payload, before passing it to the `IPacketHandler` for processing.
+
+##### ChannelHandlerContext
+The netty channel handling context that is currently processing the network connection on which the payload was received.
+This context can be used to retrieve the raw underlying connection via `ConnectionUtils` or again be used to process raw bytes of inner packets and payloads.
+
+##### Player
+This is an `Optional` containing a player.
+If the handler is invoked on the server side, then this is the player that sent the payload.
+If the handler is invoked on the client side, then this is the local player, if it is available.
+
+##### Level
+This is an `Optional` containing the level the player is in.
+If the handler is invoked on the server side, then this is the level the player is in.
+If the handler is invoked on the client side, then this is the client's current level.
```

Comment:
Are there cases where the client's level is different from the clientplayer's level? If not, this can be removed.
```suggestion
```